### PR TITLE
Support for MicoAir405Mini

### PR DESCRIPTION
**Pull request by Minderring**
Support for [MicoAir405Mini](https://micoair.com/index.php/flightcontroller_micoair405mini/).
I have fixed the issues mentioned in the previous PR and resubmitted the PR.

### CODE REVIEW

File: libraries/AP_HAL_ChibiOS/hwdef/MicoAir405Mini/hwdef.dat [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+# USART1
+PA9  USART1_TX USART1
+PA10 USART1_RX USART1
+
+# USART2 - DJIO3
+PA2 USART2_TX USART2
+PA3 USART2_RX USART2 NODMA
+
+# USART3 - GPS
```
[the remark this answers is not in the document]
Comment (Minderring) [reply libraries/AP_HAL_ChibiOS/hwdef/MicoAir405Mini/hwdef.dat:92]: All right! fixed.

**Comment by andyp1per** [comment]
LGTM, you need to squash the last commit

**Comment by Minderring** [comment]
Already squashed.

**Review by andyp1per — APPROVED** [review]
LGTM, booted ok for me

**Comment by andyp1per** [comment]
Needs the board id updating I think

**Comment by peterbarker** [comment]
@Minderring needs a rebase

**Comment by Minderring** [comment]
> Needs the board id updating I think

Okay, Board ID Updated.

**Comment by Minderring** [comment]
> @Minderring needs a rebase

Done! Please take a look.

**Comment by Minderring** [comment]
The board ID has been merged and rebased. @andyp1per @peterbarker @Hwurzburg

File: libraries/AP_HAL_ChibiOS/hwdef/MicoAir405Mini/hwdef.dat [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+PC5 LED_RED   OUTPUT LOW GPIO(0)
+PC4 LED_GREEN OUTPUT LOW GPIO(1)
+PA8 LED_BLUE  OUTPUT LOW GPIO(2)
+
+define HAL_GPIO_A_LED_PIN 0
+define HAL_GPIO_B_LED_PIN 1
+define HAL_GPIO_C_LED_PIN 2
+
+define HAL_GPIO_LED_ON 0
+define HAL_GPIO_LED_OFF 1
```
[the remark this answers is not in the document]
Comment (Minderring) [reply libraries/AP_HAL_ChibiOS/hwdef/MicoAir405Mini/hwdef.dat]: Thank you! I have removed it.

File: libraries/AP_HAL_ChibiOS/hwdef/MicoAir405Mini/README.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+![MicoAir F405 Mini V2.1 Back View](MicoAir405Mini_BackView.jpg)
+
+
+
+## UART Mapping
+
+ - SERIAL0 -> USB
+ - SERIAL1 -> UART1 (DMA-enabled)
+ - SERIAL2 -> UART2 (DJI - VTX, TX only is DMA Enabled)
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/MicoAir405Mini/README.md:28]:
```suggestion
 - SERIAL2 -> UART2 (DisplayPort, TX only is DMA Enabled)
```

File: libraries/AP_HAL_ChibiOS/hwdef/MicoAir405Mini/README.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+ - SERIAL3 -> UART3 (GPS)
+ - SERIAL4 -> UART4 (TX only is DMA Enabled)
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/MicoAir405Mini/README.md:30]:
```suggestion
 - SERIAL4 -> UART4 (MAVLink2,TX only is DMA Enabled)
```

File: libraries/AP_HAL_ChibiOS/hwdef/MicoAir405Mini/README.md [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+ - SERIAL5 -> UART5 (ESC Telemetry)
+ - SERIAL6 -> UART6 (RX6 is inverted from SBUS pin, RX only is DMA Enabled)
+
+## RC Input
+
+The default RC input is configured on the UART6_RX inverted from the SBUS pin. Other RC  protocols  should be applied at other UART port such as UART1 or UART4, and set the protocol to receive RC data: `SERIALn_PROTOCOL=23` and change SERIAL6 _Protocol to something other than '23'
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/MicoAir405Mini/README.md:36]:
```suggestion
The default RC input is configured as SBUS via the SBUS pin. Other RC  protocols  should be applied at a UART port such as UART1 or UART4, and set the protocol to receive RC data: `SERIALn_PROTOCOL=23` and change SERIAL6 _Protocol to something other than '23'
```

File: libraries/AP_HAL_ChibiOS/hwdef/MicoAir405Mini/hwdef.dat [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+define HAL_OS_FATFS_IO 1
+define HAL_BOARD_LOG_DIRECTORY "/APM/LOGS"
+define HAL_BOARD_TERRAIN_DIRECTORY "/APM/TERRAIN"
+
+# setup for OSD
+define OSD_ENABLED 1
+define HAL_OSD_TYPE_DEFAULT 1
+ROMFS_WILDCARD libraries/AP_OSD/fonts/font0.bin
+
+# reduce flash usage
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/MicoAir405Mini/hwdef.dat:148]: insteead of these...just use:
include ../include/minimize_fpv_osd.inc

File: libraries/AP_HAL_ChibiOS/hwdef/MicoAir405Mini/defaults.parm [hunk 7]
```diff
@@ -0,0 +1,11 @@
+#BATTERY MONITOR ON
+BATT_MONITOR 4
+
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/MicoAir405Mini/defaults.parm:3]:
```suggestion

#enable second OSD type
OSD_TYPE2 = 5

```

**Review by Hwurzburg — CHANGES_REQUESTED** [review]

File: libraries/AP_HAL_ChibiOS/hwdef/MicoAir405Mini/README.md (hunk 3 above)
Comment (Minderring) [reply libraries/AP_HAL_ChibiOS/hwdef/MicoAir405Mini/README.md]: Fixed.

File: libraries/AP_HAL_ChibiOS/hwdef/MicoAir405Mini/README.md (hunk 4 above)
Comment (Minderring) [reply libraries/AP_HAL_ChibiOS/hwdef/MicoAir405Mini/README.md]: Fixed.

File: libraries/AP_HAL_ChibiOS/hwdef/MicoAir405Mini/README.md (hunk 5 above)
Comment (Minderring) [reply libraries/AP_HAL_ChibiOS/hwdef/MicoAir405Mini/README.md]: Fixed.

File: libraries/AP_HAL_ChibiOS/hwdef/MicoAir405Mini/defaults.parm (hunk 7 above)
Comment (Minderring) [reply libraries/AP_HAL_ChibiOS/hwdef/MicoAir405Mini/defaults.parm:3]: Okay, added.

File: libraries/AP_HAL_ChibiOS/hwdef/MicoAir405Mini/hwdef.dat (hunk 6 above)
Comment (Minderring) [reply libraries/AP_HAL_ChibiOS/hwdef/MicoAir405Mini/hwdef.dat:148]: Got it. I have fixed it and enable OPTICALFLOW since it is important to us.

**Review by Hwurzburg — APPROVED** [review]

**Pull request merged**